src/zebra_puzzles/clue_removal.py

```diff
@@ -251,5 +257,75 @@ def remove_redundant_clues_with_solver(
         if completeness == 1:
             del chosen_clues[i]
             del chosen_constraints[i]
+            del chosen_clue_types[i]
+
+    return chosen_clues, chosen_constraints, chosen_clue_types
+
+
+def remove_red_herrings(
+    prompt: str,
+    i_red_herrings: str,
+    chosen_clue_types_str: str,
+    n_red_herrings_to_keep: int,
+) -> tuple[str, str, bool]:
+    """Remove red herrings from the list of clues.
+
+    Args:
+        prompt: The full prompt for the zebra puzzle as a string.
+        i_red_herrings: String of indices of the red herring clues in the shuffled list of clues.
+        chosen_clue_types_str: The types of clues chosen for the puzzle as a string.
+        n_red_herrings_to_keep: Number of red herring clues to keep in the prompt as an integer.
+
+    Returns:
+        A tuple (prompt, chosen_clue_types_str, fewer_rh), where:
+            prompt: The full prompt for the zebra puzzle as a string with some red herrings removed.
+            chosen_clue_types_str: The types of clues chosen for the puzzle as a string with some red herrings removed.
+            fewer_rh: Boolean indicating if fewer red herrings are included than in the original prompt.
+    """
+    # Split the string of indices into a list
+    i_red_herrings_list = i_red_herrings.split(", ")
```

Comment:
This is a bit more robust, in case a space isn't included:

```suggestion
    i_red_herrings_list = [idx.strip() for idx in  i_red_herrings.split(",")]
```